client-docs/README.md

```diff
@@ -0,0 +1,377 @@
+
+Accepts the following `opts`:
+
+- `role`: the role to grant for the invited device
+- `timeout`: the maximum amount of time in seconds to wait for a response.
+
+#### `$member.getById`
+
+`(deviceId: string) => Promise<Member>`
+
+Get the project member with the associated `deviceId`.
+
+#### `$member.getMany`
+
+`(opts: {}) => Promise<Array<Member>>`
+
+Get all project members.
+
+#### `$member.update`
+
+`(deviceId: string, info: { name?: string | null, role?: ProjectRole }) => Promise<Member>`
+
+Update information for a project member. Update is done by merging the `info` fields, as opposed to overwriting. Throws if the member does not exist or if the caller does not have the proper permissions. Resolves with the updated member information.
+
+#### `$member.remove`
+
+`(deviceId: string) => Promise<void>`
+
+Remove a member from the project. Throws if the member does not exist or if the caller does not have the proper permissions.
+
+#### Events
+
+##### `'member-update'`
+
+`$member.addEventListener('member-update', (member: Member) => void)`
+
+Emits when a new device is added to the project or information about a project member is updated.
+
+### `project.$blob`
+
+Namespace for creating blobs and getting blob info. Provides an event emitter-like interface so it can emit and subscribe to events.
+
+#### `$blob.create`
+
+`(filePaths: { original: string, preview?: string, thumbnail?: string }, metadata: { mimeType: string }) => Promise<BlobId>`
+
+Create a blob. Resolves with information about the blob that is created.
+
+Accepts the following `filePaths`:
+
+- `original`: file path to the original version of the file
+- `preview`: file path to the preview version of the file
+- `thumbnail`: file path to the thumbnail version of the file
+
+Accepts the following `metadata`:
+
+- `mimeType`: MIME type for file
+
+#### `$blob.getUrl`
+
+`(blobId: BlobId) => string`
+
+Get the http URL pointing to the desired blob. Note that this is _synchronous_ and returns the contructed URL.
+
+### DataType
+
+A DataType represents application-specific data. Each DataType has a schema that it adheres to, which can be used on the application level as well as for querying purposes at the core level. This API provides a CRUD interface to manage and work with such data.
+
+DataTypes are exposed on the client as fields on the project client instance. These fields are not prefixed with `$`. For example, accessing the interface for an application-defined "observation" type would be done so using `project.observation`. DataTypes provide an event emitter-like interface so it can emit and subscribe to events.
+
+#### Types
+
+```ts
+import { Opaque } from "type-fest";
+
+type DocId = Opaque<string>;
+type VersionId = Opaque<string>;
+
+// Represents any application-specific key-value data associated with a DataType.
+// This can be updated in the database.
+type MapeoType = {
+  [key: string]: any;
+};
+
+type DataType<T extends MapeoType> = Readonly<
```

Comment:
`DataType` -> `MapeoDoc`

`MapeoType` -> `MapeoValue`

```
type Observation = ObservationValue & MapeoCommon
```

Reply:
I think drop the generic helper, because it adds a difficult thing to name and we don't need it!